test/export/StructureDefinition.LogicalExporter.test.ts

```diff
@@ -328,6 +328,82 @@ describe('LogicalExporter', () => {
     );
   });
 
+  it('should include added element having logical model as datatype when parent is Element', () => {
+    const logicalFoo = new Logical('Foo');
+    logicalFoo.parent = 'Element';
+    const addElementRuleBars = new AddElementRule('bars');
+    addElementRuleBars.min = 0;
+    addElementRuleBars.max = '1';
+    addElementRuleBars.types = [{ type: 'Bar' }];
+    addElementRuleBars.short = 'short of property bars';
+    logicalFoo.rules.push(addElementRuleBars);
+    doc.logicals.set(logicalFoo.name, logicalFoo);
+
+    const logicalBar = new Logical('Bar');
+    const addElementRuleLength = new AddElementRule('length');
+    addElementRuleLength.min = 0;
+    addElementRuleLength.max = '1';
+    addElementRuleLength.types = [{ type: 'Quantity' }];
+    addElementRuleLength.short = 'short of property length';
+    logicalBar.rules.push(addElementRuleLength);
+    const addElementRuleWidth = new AddElementRule('width');
+    addElementRuleWidth.min = 0;
+    addElementRuleWidth.max = '1';
+    addElementRuleWidth.types = [{ type: 'Quantity' }];
+    addElementRuleWidth.short = 'short of property width';
+    logicalBar.rules.push(addElementRuleWidth);
+    doc.logicals.set(logicalBar.name, logicalBar);
+
+    const exported = exporter.export().logicals;
+    expect(exported.length).toBe(2);
+    expect(exported[0].name).toBe('Foo');
+    expect(exported[1].name).toBe('Bar');
+
+    expect(exported[0].elements).toHaveLength(4); // 1 Base element+ 2 from Element + 1 added "bars" element
+    const barsElement = exported[0].findElement('Foo.bars');
+    expect(barsElement.path).toBe('Foo.bars');
+    expect(barsElement.base.path).toBe('Foo.bars');
+    expect(barsElement.type[0].code).toBe('http://hl7.org/fhir/us/minimal/StructureDefinition/Bar');
+  });
+
+  it('should include added element having logical model as datatype when parent is another logical model', () => {
+    const logicalFoo = new Logical('Foo');
+    logicalFoo.parent = 'AlternateIdentification';
+    const addElementRuleBars = new AddElementRule('bars');
+    addElementRuleBars.min = 0;
+    addElementRuleBars.max = '1';
+    addElementRuleBars.types = [{ type: 'Bar' }];
+    addElementRuleBars.short = 'short of property bars';
+    logicalFoo.rules.push(addElementRuleBars);
+    doc.logicals.set(logicalFoo.name, logicalFoo);
+
+    const logicalBar = new Logical('Bar');
```

Comment:
To really test your logic, I think it's actually `Bar` that needs to have the parent set to `AlternateIdentification` -- because `Bar` is the type we are attempting to constrain (and get the `sdType` in the process).  If you set `logicalBar.parent` to `'AlternateIdentification'` here the test will fail.